src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,782 @@
+				this.ActualWidth >= (fe.ActualWidth * ZoomLevel) + m.OffsetX,
+				this.ActualHeight >= (fe.ActualHeight * ZoomLevel) + m.OffsetY);
+		}
+	}
+
+	private void UpdateScrollVisibility()
+	{
+		IsHorizontalScrollBarVisible = !(ContentBoundsVisibility.RightEdge && ContentBoundsVisibility.LeftEdge);
+		IsVerticalScrollBarVisible = !(ContentBoundsVisibility.TopEdge && ContentBoundsVisibility.BottomEdge);
+	}
+
+	private bool CanMoveIn((bool Horizontal, bool Vertical) _movementDirection)
+	{
+		if (ContentBoundsVisibility.AllEdgesVisible)
+		{
+			return false;
+		}
+
+		var canMove = false;
+		canMove |= CanScrollLeft() && _movementDirection.Horizontal is true;
+		canMove |= CanScrollRight() && _movementDirection.Horizontal is false;
+		canMove |= CanScrollUp() && _movementDirection.Vertical is true;
+		canMove |= CanScrollDown() && _movementDirection.Vertical is false;
+
+		return canMove;
+	}
+
+	private bool CanScrollUp() => !ContentBoundsVisibility.TopEdge;
+
+	private bool CanScrollDown() => !ContentBoundsVisibility.BottomEdge;
+
+	private bool CanScrollLeft() => !ContentBoundsVisibility.LeftEdge;
+
+	private bool CanScrollRight() => !ContentBoundsVisibility.RightEdge;
+
+	//Slide move is always on the opposite direction of the drag
+	private async void UpdateVerticalScrollBarValue()
+	{
+		VerticalScrollValue = VerticalOffset;
+		await RaiseRenderedContentUpdated();
+	}
+
+	//Slide move is always on the opposite direction of the drag
+	private async void UpdateHorizontalScrollBarValue()
+	{
+		HorizontalScrollValue = -1 * HorizontalOffset;
+		await RaiseRenderedContentUpdated();
+	}
+
+	private void IsActiveChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		if (ResetWhenNotActive && !IsActive)
+		{
+			RemoveOffset();
+			ResetZoom();
+		}
+		if (_scrollH is not null)
+		{
+			_scrollH.Visibility = IsActive ? Visibility.Visible : Visibility.Collapsed;
+		}
+		if (_scrollV is not null)
+		{
+			_scrollV.Visibility = IsActive ? Visibility.Visible : Visibility.Collapsed;
+		}
+	}
+
+	private void UpdateScrollLimits()
+	{
+		if (_presenter?.Content is FrameworkElement fe)
+		{
+			// If Actual Width/Height * Zoom < ViewPort Width/Height there should be no scroll
+			// If Actual Width/Height * Zoom > ViewPort Width/Height the amount of scroll should
+			// be based on the difference between the two values
+			var verticalScroll = Math.Max(0, (fe.ActualHeight * ZoomLevel) - ViewPortHeight);
+			var horizontalScroll = Math.Max(0, (fe.ActualWidth * ZoomLevel) - ViewPortWidth);
+
+			HorizontalMaxScroll = horizontalScroll / 2;
+			VerticalMaxScroll = verticalScroll / 2;
+
+			HorizontalMinScroll = -1 * HorizontalMaxScroll;
+			VerticalMinScroll = -1 * VerticalMaxScroll;
+		}
+	}
+
+	private void CoerceZoomLevel(DependencyObject sender, DependencyProperty dp)
+	{
+		var control = (ZoomContentControl)sender;
+		control.ZoomLevel = Math.Clamp(control.ZoomLevel, control.MinZoomLevel, control.MaxZoomLevel);
+	}
+
+	public ZoomContentControl()
```

Comment:
ctor and OnApplyTemplate should be on the very top of this file, only after static members, fields and properties